capa/features/extractors/dnfile/helpers.py

```diff
@@ -107,8 +107,18 @@ def format_name(module, method):
         return f"{module}.{method}"
 
 
+def validate_has_dotnet(pe: dnfile.dnPE):
+    assert pe.net is not None
+    assert pe.net.mdtables is not None
+    assert pe.net.Flags is not None
+
+
 def resolve_dotnet_token(pe: dnfile.dnPE, token: Token) -> Any:
     """map generic token to string or table row"""
+    validate_has_dotnet(pe)
+    assert pe.net is not None
+    assert pe.net.mdtables is not None
```

Comment:
mypy doesn't follow the validation in `validate_has_dotnet()` so really that function call doesn't do anything for mypy. initially i thought it might, but no. there is duplication.

however, it seemed like a good place to put common dotnet validation and has a very readable name. if we wanted to handle missing properties and raise a specific exception, doing it within `validate_has_dotnet` would be a good way.

as i write this comment, i'm having a hard time describing why i think it needs to be here, so thats a good clue its not needed, so i'll remove :-)

Reply:
removed in [505910e](https://github.com/mandiant/capa/pull/1235/commits/505910edb5a46215fd936d844adfd3fa74761a3b)